src/test/test_proj_data_info_subsets.cxx

```diff
@@ -373,28 +452,52 @@ test_forward_projection_for_one_subset(
       // loop over all segments to check viewgram for all segments
       for (int segment_num = full_forward_projection.get_min_segment_num(); segment_num < full_forward_projection.get_max_segment_num(); ++segment_num)
         {
-          check_if_equal(
-                          full_forward_projection.get_viewgram(subset_views[i], segment_num),
-                          subset_forward_projection_uptr->get_viewgram(i, segment_num), "Are viewgrams equal?");
+          if (!check_if_equal(
+            full_forward_projection.get_viewgram(subset_views[i], segment_num),
+            subset_forward_projection_uptr->get_viewgram(i, segment_num), "Are viewgrams equal?"))
+          {
+            cerr << "testing forward projection failed: viewgrams weren't equal in subset " << i << endl;
+            break;
+          }
           // TODO also compare viewgram metadata
         }
     }
 }
 
 
-// void TestProjDataInfoSubsets::
-// test_back_projection_is_consistent(
-//     const ProjData &input_sino, const DiscretisedDensity<3,float> &back_projection,
-//     const BackProjectorByBin& bck_projector, int num_subsets)
-// {
-//     DiscretisedDensity<3,float> back_projection_sum = back_projection.clone();
-//     back_projection_sum.fill(0.f);
+void TestProjDataInfoSubsets::
+test_back_projection_is_consistent(
+    const shared_ptr<const ProjData>& input_sino_sptr,
+    const shared_ptr<const VoxelsOnCartesianGrid<float> >& template_image_sptr,
+    int num_subsets)
+{
+  auto full_back_projection = generate_full_back_projection(
+    input_sino_sptr, template_image_sptr, /*use_syummetries=*/false);
 
-//     for (int s=0; s++; s<num_subsets) {
-//         ProjData& subset = *input_sino.get_subset(s, num_subsets);
+  VoxelsOnCartesianGrid<float> back_projection_sum = *template_image_sptr->clone();
+  back_projection_sum.fill(0.f);
 
-//     }
-// }
+  for (unsigned int subset_n=0; subset_n<num_subsets; ++subset_n) {
+    auto subset_views = _calc_regularly_sampled_views_for_subset(
+      subset_n, num_subsets, input_sino_sptr->get_num_views());
+
+    ProjData& subset = *input_sino_sptr->get_subset(subset_views);
+
+    auto subset_back_projector_sptr = construct_projector_pair(
+      input_sino_sptr->get_proj_data_info_sptr(), template_image_sptr, /*use_symmetries=*/false)
+      ->get_back_projector_sptr() ;
+
+    VoxelsOnCartesianGrid<float> subset_back_projection = *template_image_sptr->clone();
+    subset_back_projector_sptr->back_project(subset_back_projection, *input_sino_sptr);
+    back_projection_sum += subset_back_projection;
+  }
+  back_projection_sum /= num_subsets;  // Why do I have to do this...?!
```

Comment:
@KrisThielemans  could you please check this line? I expected when I wrote the test that I would sum the subsets, but I instead averaged them to run the test. Hopefully I just misunderstood the backproj units (activity vs counts)

Reply:
I can't check the code at the moment, but backprojection sums. a problem with using `template_image_sptr->clone()`? Better to guarantee it starts with zero (i.e. use `get_uniform_copy` I guess)

Reply:
Disregard, this is my mistake. I backproject the original image every time, not the subset!!

Reply:
(but I need to go to bed now, I've pushed but its producing a segfault. I'll look tomorrow afternoon)